deeplinkdispatch/src/main/java/com/airbnb/deeplinkdispatch/BaseDeepLinkDelegate.java

```diff
@@ -130,28 +134,35 @@ public DeepLinkResult dispatchFrom(Activity activity, Intent sourceIntent) {
         } else {
           activity.startActivity(newIntent);
         }
-        return createResultAndNotify(activity, true, uri, entry.getUriTemplate(), null);
+        return createResultAndNotify(activity, true, uri, entry.getUriTemplate(), null,
+                newIntent.getComponent().toString());
       } catch (NoSuchMethodException exception) {
         return createResultAndNotify(activity, false, uri,
-                entry.getUriTemplate(), "Deep link to non-existent method: " + entry.getMethod());
+                entry.getUriTemplate(), "Deep link to non-existent method: " + entry.getMethod(),
+               "");
       } catch (IllegalAccessException exception) {
         return createResultAndNotify(activity, false, uri,
-                entry.getUriTemplate(), "Could not deep link to method: " + entry.getMethod());
+                entry.getUriTemplate(), "Could not deep link to method: " + entry.getMethod(),
```

Comment:
Instead of the string for the error, we could pass the exception. (Again, that is the most accurate representation of the result.)